meetings/2016-12-01.md

```diff
@@ -0,0 +1,71 @@
+## Version Management and Installers Discussion
+
+* Date: 2016-12-01 (Collaboration Summit - Austin)
+* GitHub issue: <https://github.com/nodejs/version-management/issues/5>
+* Minutes: <https://docs.google.com/document/d/1UDDnHrmuVdG-ih3-Q8m5hbCfE7z7IuaVoB0keMC9as4>
+
+## Attendees
+
+* Corey Butler (@coreybutler)
+* Myles Borins (@thealphanerd)
+* Josh Gavant (@joshgav)
+* William Kapke (@williamkapke)
+* Mikeal Rogers (@mikeal)
+* Jeremiah Senkpiel (@fishrock123)
+*
+
+**Next steps:**
+
+* Document and come to consensus on default patterns, paths, etc.
+* Investigate cross-platform installer options like Electron.
+* Track usage of different installation methods on website, perhaps via a
+  header.
+* Add a guide to the web site on available version managers and how they work.
+
+---
+
+What are the use cases for version managers?
+
+* Developer installing on a local machine, we don’t want to force them to the
+  command line.
+* Developer who needs to switch environments for testing.
+* Assist with updates to runtime.
+
+Installation is our most important use case, enable people to get started
+quickly from the web site home page.
+
+Some systems have their own system package manager (apt, homebrew). But the
+package author would bypass any of our systems and use the raw tarballs.
+
+Should we orient around the needs of the installer? What might that mean?
+
+* Would be best if the installer utilized the command-line/API packages.
+* Consider Electron installer?
+    * Initial download is 136MB, but may be possible to trim.
+    * Work has begun at [nodejs/installer](https://github.com/nodejs/installer).
+* Are there alternatives?
+    * bitRock
+    * Qt?
+    * Web page with config options which generates a script which can be run
+      automatically.
+
+Should we focus on a JS implementation?
+
+* Would encourage more contributors, but is that important?
+* nvm uses POSIX. Best part is that it doesn’t require JS or a Node runtime. But
+  it doesn’t work on Win32 and it’s more difficult to modify.
+* Maybe we should focus on Windows independently anyway? JS code would also be
+  full of “#ifdef”-like statements.
```

Comment:
[NVS](https://github.com/jasongin/nvs) is a cross-platform JS version-manager that is not full of "#ifdef"-like statements. While there certainly are *some* platform-specific sections of code, the great majority of the implementation can be cross-platform, thanks to the cross-platform APIs provided by the Node.js core modules.

Another significant consideration not listed here is that two separately-developed tools are unlikely to provide  consistent *functionality* and *user experience* across platforms.